### PR TITLE
Disable script editing for those not in scripts.yaml

### LINKED ISSUE #3563: Script panel should show uneditable scripts in disabled state
**Home Assistant release with the issue:**
0.98.1

**Description of problem:**

In the Automation config panel, it helpfully displays the pencil "edit" icons in a grayed out disabled state for uneditable automations:

automations

However that same helpful feature is not currently applied to the Script config panel. All of the edit buttons are shown in the primary text color and _appear_ to be editable until you actually click on one and get an error.

scripts

error

Side note: Scripts section is also missing the "last triggered" secondary text and "info" button that Automations section has.

### PR DESCRIPTION
**Pull request by iantrich**
Closes https://github.com/home-assistant/home-assistant-polymer/issues/3563

### CODE REVIEW

File: src/panels/config/script/ha-script-picker.ts [hunk 1]
```diff
@@ -68,10 +69,25 @@ class HaScriptPicker extends LitElement {
...
+                              : undefined
+                          )}
+                        >
                           <paper-icon-button
                             icon="hass:pencil"
+                            .disabled=${!script.attributes.id}
                           ></paper-icon-button>
+                          ${!script.attributes.id
+                            ? html`
+                                <paper-tooltip position="left">
```
Comment (bramkragten) [inline src/panels/config/script/ha-script-picker.ts]: You need to import `paper-tooltip`
`import "@polymer/paper-tooltip/paper-tooltip";`

**Comment by iantrich** [comment]
last commit is not working, actually. I didn't clear my cache properly while testing. trying to recreate changes I forgot to commit last night.

**Comment by bramkragten** [comment]
Not sure about this, but think we might want to solve this the same way as the automation is done, although I'm not sure how that works, as it only checks if `id` is preset, which could be the case outside of `automations.yaml`? Otherwise we should create a new backend command so we can get all the scripts at once instead of creating a call for every entity

**Comment by iantrich** [comment]
Yeah, new backend call is probably the route needed

**Comment by balloob** [comment]
`render()` should never have any side effects. `render()` will also be called for every state change, unrelated to scripts. Now imagine you have 100 scripts, a sensor changes it's value and we make a 100 API requests! 

We should probably do a similar thing as we do with `id`. The reason scripts have no `id` is because the configuration is keyed by entity ID, while automations are a list (and so we need an ID to identify them)

**Comment by balloob** [comment]
A backend call can work too.

**Comment by iantrich** [comment]
@balloob so I'm thinking we just close this PR and open an issue on the backend so that when we retrieve scripts we can tell if the they are editable or not without an additional call.

**Comment by balloob** [comment]
Sounds good.

But who will do that issue 🙈

**Comment by iantrich** [comment]
@balloob make the new guy do it 😜

**Pull request closed without merging**